src/main/java/qna/service/QnAService.java

```diff
@@ -35,24 +35,15 @@ public Question findQuestionById(Long id) {
     @Transactional
     public void deleteQuestion(User loginUser, long questionId) throws CannotDeleteException {
         Question question = findQuestionById(questionId);
-        if (!question.isOwner(loginUser)) {
-            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
-        }
-
-        List<Answer> answers = question.getAnswers();
-        for (Answer answer : answers) {
-            if (!answer.isOwner(loginUser)) {
-                throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
-            }
-        }
-
-        List<DeleteHistory> deleteHistories = new ArrayList<>();
-        question.setDeleted(true);
-        deleteHistories.add(new DeleteHistory(ContentType.QUESTION, questionId, question.getWriter(), LocalDateTime.now()));
-        for (Answer answer : answers) {
-            answer.setDeleted(true);
-            deleteHistories.add(new DeleteHistory(ContentType.ANSWER, answer.getId(), answer.getWriter(), LocalDateTime.now()));
-        }
-        deleteHistoryService.saveAll(deleteHistories);
+        question.delete(loginUser);
+
+        Answers answers = question.getAnswers();
+        answers.delete(loginUser);
+
+        DeleteHistories deleteHistories = new DeleteHistories();
+        deleteHistories.add(question);
+        deleteHistories.add(answers);
```

Comment:
```suggestion
        DeleteHistories deleteHistories = new DeleteHistories();
        question.record(deleteHistories);
        answers.record(deleteHistories);
```

저라면 DeleteHistories 의 public 메서드를 늘려서,
테스트해야하는 책임과 역할을 늘리기 보다는,
기존의 객체를 그냥 그대로 활용하려 할 것 같습니다.